src/repositories/user.repository.ts

```diff
@@ -50,4 +50,26 @@ export class UserRepository {
     });
     return productList;
   }
+  public async getProductById(index: number): Promise<IProduct> {
+    const productFound = await this.db.query({
+      sql: `SELECT * FROM products WHERE productId= ${index}`,
```

Comment:
Check if it is possible to use a params property on this function, check what brings the driver from the driver.
This kind of instruction is Highly insecure as could lead to an SQL Injection!

Reply:
Hi ! not very sure what you mean, I change the * to 
SELECT productId,name, description, imagePath, price, created, sold, id FROM products
Is this what you mean?
In the users feature, we implemented it that way.

Reply:
I'm not sure how to do it.  is for research.  but certainly, this is a serious fault to security.  this kind of instruction on code is a target for a SQL Injection attack.

Reply:
I"m talking about the way you interpolate strings. without any kind of filter. for the string that comes from the client.

Reply:
that allows anyone to inject, I mean send prepared kind of injection, vectors into your database.

Reply:
Done, please have a look. Thanks